nix/lib/rust.nix

```diff
@@ -7,7 +7,7 @@ let
 in
 rec {
   rust_default = { override ? { } }: rec {
-    nightly_pkg = pkgs.rust-bin.nightly."2021-11-22";
+    nightly_pkg = pkgs.rust-bin.nightly."2022-06-22";
```

Comment:
Is it worth to make this change for data-plane as well? Having same versions for both would ease development.

Reply:
Yep, it'd be good to get the latest clippy suggestions.